### PR TITLE
Fix incremental working set on out of tree subjects

### LINKED ISSUE #382: Out of repository subjects under incremental mutation run
When running with the `--since` flag subjects outside the repository are being flagged as changed.

For example, when running against spree, the `/home/ubuntu/spree/vendor/bundle/ruby/2.1.0/gems/paranoia-2.1.3/lib/paranoia.rb` file is being identified as changed, which is causing an error since it's outside the git repo.

### PR DESCRIPTION
**Pull request by mbj**
[fix #382]

### CODE REVIEW

**Comment by mbj** [comment]
This was validated via:

```
bundle exec mutant --zombie -I lib -r mutant --use rspec --since master -- 'Mutant*'
```

**Comment by mbj** [comment]
@dkubb Assuming all green this is ready for review.

After merging, #383 this branch will be in conflict with the expected flay score, it would be great if you could fix this in case this is the only merge blocker.

**Comment by mbj** [comment]
@dkubb Assuming all green this is also ready for the next pass.

File: lib/mutant/repository.rb [hunk 1]
```diff
@@ -61,6 +63,21 @@ def touches?(path, line_range)
...
+      #
+      # @param [Pathname] path
+      #
+      # @return [Boolean]
+      #
+      # @api private
+      def tracks?(path)
+        command = %W[git ls-files --error-unmatch -- #{path}]
+        _stdout, _stderr, status = Open3.capture3(*command, binmode: true)
+        status.success?
```
Comment (dkubb) [inline lib/mutant/repository.rb]: For consistency, I wonder if the other method in this class should omit blank lines between similar statements as this method?

Comment (mbj) [inline lib/mutant/repository.rb]: The other method has 2 branches. This one not.

I added the blank lines in the other method to make sure the branches are not "buried" in the code.

Comment (mbj) [inline lib/mutant/repository.rb]: I actually think that we should use `Process.spawn` here and not "bend" the `open3` into what we need, via ignoring stdout, and stderr. I'll fix that.

Comment (dkubb) [inline lib/mutant/repository.rb]: Why not just `Kernel.system` since all you care about is the return value?

**Comment by mbj** [comment]
@dkubb This is ready for the next pass.

File: lib/mutant/repository.rb [hunk 2]
```diff
@@ -61,6 +63,26 @@ def touches?(path, line_range)
...
+      # @param [Pathname] path
+      #
+      # @return [Boolean]
+      #
+      # @api private
+      def tracks?(path)
+        command = %W[git ls-files --error-unmatch -- #{path}]
+        Kernel.system(
+          *command,
+          out: File::NULL,
```
Comment (dkubb) [inline lib/mutant/repository.rb]: Oh, this is really nice, I didn't know these options existed.

Comment (mbj) [inline lib/mutant/repository.rb]: It reduces the need to shell out, and build shell commands. Thats great as no escaping is required. (Still beware for option injection).

**Pull request merged**